pydomcfg/domzgr/zgr.py

```diff
@@ -0,0 +1,164 @@
+            *) W-points are at integer values - 1/2 (between 0.5 and jpk-0.5)
+        grd: str
+            If we are dealing with "T" or "W" model levels.
+
+        Returns
+        -------
+        ps: float
+            Uniform non-dimensional sigma-coordinate (0. <= sigma <= -1)
+        """
+
+        kindx = float(k + 1)  # to deal with python convention
+        if grd == "W":
+            kindx -= 0.5
+
+        ps = -(kindx - 0.5) / float(self._jpk - 1)
+        return ps
+
+    # -------------------------------------------------------------------------
+    @staticmethod
+    def compute_z3(
+        su: float,
+        ss1: float,
+        a1: float,
+        a2: float,
+        a3: float,
+        ss2: float = 0.0,
+        a4: float = 0.0,
+    ):
+        """
+        Generalised function providing the analytical
+        transformation from computational space to
+        physical space. It takes advantage of the fact that
+        z-coordinates can be considered a special case of
+        s-coordinate.
+
+        Note that z is downward positive.
+
+        Parameters
+        ----------
+        su: float
+            uniform non-dimensional vertical coordinate s,
+            aka sigma-coordinates. 0 <= s <= 1
+        ss1: float
+            stretched non-dimensional vertical coordinate s,
+            0 <= s <= 1
+        a1: float
+            parameter of the transformation
+        a2: float
+            parameter of the transformation
+        a3: float
+            parameter of the transformation
+        ss2: float
+            second stretched non-dimensional vertical coordinate s,
+            0 <= s <= 1 (only used for zco with ldbletanh = True)
+        a4: float
+            parameter of the transformation (only used for zco with ldbletanh = True)
+
+        Returns
+        -------
+        z: float
+            Depths of model levels
+        """
+
+        z = a1 + a2 * su + a3 * ss1 + a4 * ss2
+        return z
+
+    # -------------------------------------------------------------------------
+    def compute_e3(self, ds: Dataset):
+        """
+        Grid cell thickness computed as discrete derivative
+        (central-difference) of levels' depth
+
+        Parameters
+        ----------
+        ds: Dataset
+            xarray dataset with empty ``e3{T,W}`` and ``z3{T,W}``
+            correctly computed
+        Returns
+        -------
+        ds: Dataset
+            xarray dataset with ``e3{T,W}`` correctly computed
+        """
+        for k in range(self._jpk - 1):
+            ds["e3T"][{"z": k}] = ds["z3W"][{"z": k + 1}] - ds["z3W"][{"z": k}]
+            ds["e3W"][{"z": k + 1}] = ds["z3T"][{"z": k + 1}] - ds["z3T"][{"z": k}]
+        # Bottom:
+        k = -1
+        ds["e3T"][{"z": k}] = 2.0 * (ds["z3T"][{"z": k}] - ds["z3W"][{"z": k}])
+        # Surface:
+        k = 0
+        ds["e3W"][{"z": k}] = 2.0 * (ds["z3T"][{"z": k}] - ds["z3W"][{"z": k}])
```

Comment:
They are doing the same thing, how about:
```suggestion
        for varname, k in zip(["e3T", "e3W"], [-1, 0]):
            ds[varname][{"z": k}] = 2.0 * (ds["z3T"][{"z": k}] - ds["z3W"][{"z": k}])
```